### ISSUE TITLE
[Bug]: IgnoredArticles/ND_IGNOREDARTICLES Does Not Seem to Be Working

### ISSUE THREAD

**Issue by dathbe** [navidrome/navidrome#2819]
### I confirm that:

- [X] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [X] I'm using the latest version (your issue may have been fixed already)

### Version

0.51.0 (fd61b29a)

### Current Behavior

Songs beginning with, e.g., "A" are sorted with the As instead of with the letter beginning the next word.  This is true whether I explicitly set ND_IGNOREDARTICLES in Docker or rely on the documented default behavior.  (See [here](https://www.navidrome.org/docs/usage/configuration-options/#:~:text=Empty%20(disabled)-,IgnoredArticles,-ND_IGNOREDARTICLES))

![2024-01-26 16_04_54-Navidrome — LibreWolf](https://github.com/navidrome/navidrome/assets/14321825/1db4cd6c-d907-414b-a34f-6ff6d682c82f)

### Expected Behavior

The software should ignore common articles.

### Steps To Reproduce

Sort any list.

### Environment

```markdown
- OS: Docker running in Debian
- Browser: Firefox
- Client: Firefox
```

### How Navidrome is installed?

Docker

### Configuration

```toml
n/a, but I'd be happy to share if someone finds it useful.
```

### Relevant log output

```shell
none
```

### Code of Conduct

- [X] I agree to follow Navidrome's Code of Conduct

**Comment by deluan** [comment]
This configuration is only meant to be used for Artists and Albums names. For Song titles, the only solution is switch to use `Sort` tags and configure `PreferSortTags=true`. Of course this will also affect Artists and Albums, but there's not much that can be done about it. 

Unfortunately there's no silver bullet solution. See also discussion here: https://github.com/navidrome/navidrome/issues/174

**Comment by dathbe** [comment]
It's not working for artists either:

![2024-01-26 19_28_16-Navidrome — LibreWolf](https://github.com/navidrome/navidrome/assets/14321825/43dab004-2f75-4694-a387-84df578d5209)

**Comment by deluan** [comment]
It definitely works with artists. See screenshot from our `demo` server:
![Screenshot 2024-01-27 at 11 47 45 AM](https://github.com/navidrome/navidrome/assets/331353/7afc1ccd-a66f-4ce6-93fe-f9ed042116b3)

When changing the `IgnoredArticles` param, you need to do a full scan for the change to be effective, as the articles removal is applied when importing tracks to the DB.

**Comment by dathbe** [comment]
It's not working in playlists in your demo server:
![image](https://github.com/navidrome/navidrome/assets/14321825/1d1180e7-aefe-427f-b45a-9a37d912d67e)

**Comment by deluan** [comment]
Ah, context :) Yes, you are right, it is not working when sorting playlists by artists. I'll take a look at that. Thanks for reporting.

**Comment by dathbe** [comment]
Thanks!

**Comment by dathbe** [comment]
Confirmed this has been fixed in 0.51.1.
